config.m4

```diff
@@ -217,7 +217,6 @@ if test "$MONGODB" != "no"; then
     bson-timegm.c \
     bson-utf8.c \
     bson-value.c \
-    bson-version.c \
```

Comment:
In this case, we also need the `mongoc-version-functions.c` added to `MONGOC_SOURCES` (it was never included). I'll fix both.